sklearn/utils/tests/test_validation.py

```diff
@@ -1153,3 +1153,27 @@ def test_check_fit_params(indices):
         result['sparse-col'],
         _safe_indexing(fit_params['sparse-col'], indices_)
     )
+
+
+def test_check_sparse_pandas_default():
+    pd = pytest.importorskip("pandas")
+    sp_mat = _sparse_random_matrix(10, 3)
+
+    sdf = pd.DataFrame.sparse.from_spmatrix(sp_mat)
+    result = check_array(sdf, accept_sparse=True)
+
+    # by default pandas converts to coo
```

Comment:
Is this pandas default, or is it rather what we did in `check_array`?

Reply:
> Is this pandas default, or is it rather what we did in check_array?

Both -- pandas only have direct conversion to coo at the moment, so it makes sense to keep that as default. Conversions from COO to either CSC or CSR is fast.